packages/modules/viewer-react/src/components/iModel/IModelLoader.tsx

```diff
@@ -184,35 +181,17 @@ const Loader: React.FC<ModelLoaderProps> = React.memo(
             onIModelConnected(imodelConnection);
           }
 
-          const viewIds = await getDefaultViewIds(imodelConnection);
-
-          if (viewIds.length === 0 && contextId && iModelId) {
-            // no valid view data in the model. Direct the user to the synchronization portal
-            const msgDiv = document.createElement("div");
-            const msg = await BaseInitializer.getIModelDataErrorMessage(
-              contextId,
-              iModelId,
-              IModelApp.i18n.translateWithNamespace(
-                "iTwinViewer",
-                "iModels.emptyIModelError"
-              )
-            );
-            msgDiv.innerHTML = msg;
-            // this can and should be async. No need to wait on it
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            IModelApp.notifications.openMessageBox(
-              MessageBoxType.Ok,
-              msgDiv,
-              MessageBoxIconType.Critical
-            );
-          }
-
           // attempt to construct a default viewState
-          const savedViewState = await ViewCreator.createDefaultView(
-            imodelConnection,
-            undefined,
-            viewIds.length > 0 ? viewIds[0] : undefined
-          );
+          const viewCreator = new ViewCreator3d(imodelConnection);
+
+          const options: ViewCreator3dOptions = viewCreatorOptions
```

Comment:
Since you set useSeedView in the conditional below couldn't you just do:

```ts
const options: ViewCreator3dOptions = viewCreatorOptions ?? {};
```

Reply:
could sure, but why not initialize the value when we know it will be true in that case? the setter below was for the condition where the user passes options, but no useSeedView. Also didn't want to mutate a prop, thus the spread operator